packages/dx-grid-core/src/plugins/integrated-selection/computeds.js

```diff
@@ -6,11 +6,17 @@ export const getAvailableToSelect = (rows, getRowId, isGroupRow) => {
   return dataRows.map(row => getRowId(row));
 };
 
-export const someSelected = ({ selection, availableToSelect }) =>
-  availableToSelect.length !== 0 && selection.size !== 0
-    && availableToSelect.some(elem => selection.has(elem))
-    && availableToSelect.some(elem => !selection.has(elem));
+export const someSelected = ({ selection, availableToSelect }) => {
+  const selectionSet = new Set(selection);
```

Comment:
I am note sure that we should do this, but it will be better to convert selection only if the `availableToSelect.length !== 0 && selectionSet.size !== 0` condition is true.